util/profile/sim65-prof.c

```diff
@@ -0,0 +1,189 @@
+/*****************************************************************************/
+/* sim65-prof                                                                */
+/* (C) 2023 Lauri Kasanen                                                    */
+/*                                                                           */
+/* This software is provided 'as-is', without any expressed or implied       */
+/* warranty.  In no event will the authors be held liable for any damages    */
+/* arising from the use of this software.                                    */
+/*                                                                           */
+/* Permission is granted to anyone to use this software for any purpose,     */
+/* including commercial applications, and to alter it and redistribute it    */
+/* freely, subject to the following restrictions:                            */
+/*                                                                           */
+/* 1. The origin of this software must not be misrepresented; you must not   */
+/*    claim that you wrote the original software. If you use this software   */
+/*    in a product, an acknowledgment in the product documentation would be  */
+/*    appreciated but is not required.                                       */
+/* 2. Altered source versions must be plainly marked as such, and must not   */
+/*    be misrepresented as being the original software.                      */
+/* 3. This notice may not be removed or altered from any source              */
+/*    distribution.                                                          */
+/*                                                                           */
+/*****************************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+static unsigned samples[64 * 1024];
+
+struct label_t {
+    char name[32];
+    unsigned short addr;
+    unsigned count;
+};
+
+#define MAX_LABELS 1024
+static struct label_t labels[MAX_LABELS];
+static unsigned numlabels;
+
+static int byaddr(const void *ap, const void *bp)
+{
+    const struct label_t *a = (struct label_t *) ap;
+    const struct label_t *b = (struct label_t *) bp;
+
+    if (a->addr < b->addr)
+        return -1;
+    if (a->addr > b->addr)
+        return 1;
+    return 0;
```

Comment:
please always use braces for if statements

Reply:
or `return a->addr > b->addr;`

Reply:
That would be incorrect, it's not a boolean return value. Perhaps you meant `return a->addr - b->addr`?